subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -272,9 +276,15 @@ public void visitInputProperty(String propertyName, PropertyValue value, boolean
                     Object preparedValue = InputParameterUtils.prepareInputParameterValue(value);
 
                     if (preparedValue == null && !optional) {
-                        validationContext.visitPropertyProblem(WARNING,
-                            propertyName,
-                            "does not have a value specified"
+                        validationContext.visitPropertyProblem(problem ->
+                            problem.withId(ValidationProblemId.VALUE_NOT_SET)
+                                .reportAs(Severity.ERROR)
+                                .forProperty(propertyName)
+                                .withDescription("doesn't have a configured value")
+                                .happensBecause("This property isn't marked as optional and no value has been configured")
```

Comment:
Is there some value to share to implementation of this error message between `DefaultTransformer` and `AbstractValidatingProperty`?